packages/@n8n/ai-workflow-builder.ee/src/tools/utils/node-configuration.utils.ts

```diff
@@ -0,0 +1,146 @@
+import type { INodeParameters } from 'n8n-workflow';
+
+import { MAX_NODE_EXAMPLE_CHARS } from '@/constants';
+import type { NodeConfigurationsMap, NodeConfigurationEntry, WorkflowMetadata } from '@/types';
+
+/**
+ * Node structure for configuration collection
+ */
+interface NodeForConfiguration {
+	type: string;
+	typeVersion: number;
+	parameters: INodeParameters;
+}
+
+/**
+ * Collect configuration from a single node if it meets size requirements
+ * Returns null if the node has no parameters or exceeds size limits
+ */
+export function collectSingleNodeConfiguration(
+	node: NodeForConfiguration,
+): NodeConfigurationEntry | null {
+	const hasParams = Object.keys(node.parameters).length > 0;
+	if (!hasParams) return null;
+
+	const parametersStr = JSON.stringify(node.parameters);
+	if (parametersStr.length > MAX_NODE_EXAMPLE_CHARS) return null;
```

Comment:
I wonder if we will lose a lot of agents with long system prompts this way. Could we trim the parameter itself instead?